src/main/java/Storage.java

```diff
@@ -0,0 +1,101 @@
+package duke;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import duke.task.Task;
+import duke.task.ToDo;
+import duke.task.Event;
+import duke.task.Deadline;
+
+/**
+ * Represents a storage which stores the Task List.
+ * The storage is read at when the bot starts up and written into when the bot is terminated.
+ */
+
+public class Storage {
+    private String filePath;
+
+    /**
+     * Creates a Storage Object containing the file path of where the user wishes to store the TaskList.
+     * @param filePath file path of Storage.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Reads and loads the storage into an ArrayList of Tasks.
+     * @return ArrayList of tasks stored in the storage.
+     * @throws FileNotFoundException if file path does not exist.
+     */
+
+    public ArrayList<Task> load() throws FileNotFoundException {
```

Comment:
I think there is too much nesting in this method which makes the code hard to follow. Consider putting some of the if-else parts into seperate methods in order to simplify it.